### ISSUE TITLE
Can ContentProtection default_KID overwrite tenc box in fmp4?

### ISSUE THREAD

**Issue by aoranli** [google/shaka-player#1783]
Hi,

I have generated a fmp4 file using Apple's `mediafilesegmenter` and it is having `0x00000000000000000000000000000000` as the key identifier in the `tenc` box. However, I would like to use a different kid so I specified the default_KID in the mpd file like this:

```
<ContentProtection value="cbcs" schemeIdUri="urn:mpeg:dash:mp4protection:2011"  cenc:default_KID="abba271e-8bcf-552b-bd2e-86a434a9a5d9"/>
```

I have to use the all 0 keyid in my player configuration to make the decryption work otherwise it won't work.

```
  player.configure({
    drm: {
      clearKeys: {
       '00000000000000000000000000000000': my_key,
  });
```
Does shaka player allow this?

Thank you.

**Comment by TheModMaker** [comment]
No, the `default_KID` must match the value in the `tenc` box.  We mainly use this field to determine whether we have the correct keys; if we don't have the `default_KID` key, we'll not play that stream assuming the browser can't decrypt it.  There is no way to change the value of the `tenc` box since that is what the browser uses to find the correct key to decrypt with.
